presto-docs/src/main/sphinx/sql/grant.rst

```diff
@@ -0,0 +1,51 @@
+=====
+GRANT
+=====
+
+Synopsis
+--------
+
+.. code-block:: none
+
+    GRANT ( privilege [, ...] | ( ALL PRIVILEGES ) )
+    ON [ TABLE ] table_name TO ( grantee | PUBLIC )
+    [ WITH GRANT OPTION ]
+
+Usage of the term ``grantee`` denotes both users and roles.
+
+Description
+-----------
+
+Grants the specified privileges to the specified grantee.
+
+Specifying ``ALL PRIVILEGES`` grants ``DELETE, INSERT`` and ``SELECT`` privileges.
+
+Specifying ``PUBLIC`` grants privileges to all grantees.
+
+The optional ``WITH GRANT OPTION`` clause allows the grantee to grant these same privileges to others.
+
+For ``Grant`` statement to succeed, the user executing it should possess the specified privileges as well as the ``GRANT OPTION`` for those privileges.
+
+Limitation
+----------
+
+``GRANT`` is currently supported only for hive connector.
```

Comment:
That's generally true anytime a new feature is added: we add it to the engine and to the first connector (which was the original motivation / driver for the feature). Then we later add support in more connectors.

Connector specific information with the SQL syntax quickly becomes out of date, or is confusing for users with custom connectors. For example, someone might have an private connector that does support the feature.

We could add a "Limitations" section similar to DELETE (I personally dislike that but understand the need for it).